pkg/apis/core/v1alpha1/utils.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2020 The CDI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PopulationComplete indicates if the persistent volume passed in has been fully populated. It follow the following logic
+// 1. If the PVC is not owned by a DataVolume, return true, we assume someone else has properly populated the image
+// 2. If the PVC is owned by a DataVolume, look up the DV and check the phase, if phase succeeded return true
+// 3. If the PVC is owned by a DataVolume, look up the DV and check the phase, if phase !succeeded return false
+func PopulationComplete(pvc *corev1.PersistentVolumeClaim, getDvFunc func(name, namespace string) (*DataVolume, error)) (bool, error) {
```

Comment:
I wonder about the usage of this on KubeVirt side.  Specifically the case where a PVC is owned by a DV but population hasn't completed yet.

I imagine the KubeVirt side would be something like this:

```go
dvOwner, err := getDataVolumeOwner(pvc)
if err != nil{
   ...
}

if dvOwner != nil && dvOwner.Status != Succeeded {
   // wait until succeeded
}

// continue starting vm

```

if the usage above makes sense, then I'm not sure it really makes sense to vendor in a function over there

Reply:
whoops my bad